### PR TITLE
RUBY-2655 Fix GH actions

### PR DESCRIPTION
**Pull request by johnnyshields**
(GH Actions is already installed in the repo)

### CODE REVIEW

**Comment by comandeo** [comment]
https://jira.mongodb.org/browse/RUBY-2655
* Fix GH actions
* Remove references to travis and appveyor

**Review by comandeo — APPROVED** [review]

File: .github/workflows/bson-ruby.yml [hunk 1]
```diff
@@ -20,15 +20,13 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu, macos, windows ]
-        ruby: [ 2.3, 2.4, 2.5, 2.6, 2.7, head ]
-        include:
-          - { os: windows , ruby: mingw }
```
Comment (p-mongo) [inline .github/workflows/bson-ruby.yml:25]: Why was windows removed please?

Comment (comandeo) [reply .github/workflows/bson-ruby.yml:25]: Stable rubies on windows are still included. I was not sure whether we support `mingw` version. Do we need it?

Comment (johnnyshields) [reply .github/workflows/bson-ruby.yml:25]: should be mingw64, and yes, please include this! :)

Comment (johnnyshields) [reply .github/workflows/bson-ruby.yml:25]: (It might be `x64_mingw` actually.)

Comment (johnnyshields) [reply .github/workflows/bson-ruby.yml:25]: However this is the Ruby "platform", not the ruby "version"

Comment (comandeo) [reply .github/workflows/bson-ruby.yml:25]: mingw is back, as well as ruby head for macos and linux

**Review by p-mongo — APPROVED** [review]
:+1: Thanks!

**Comment by johnnyshields** [comment]
@comandeo "windows mingw" doesn't make as a ruby version... mingw is a platform like darwin is for mac. I think there should be a different way to do this (we want to use `x64_mingw`, not the 32 bit one).

Also, should check if we are testing on ARM as well for linux and ideally mac.

**Pull request merged**